### ISSUE TITLE
test message sent out ok, but contact form details don't send

### ISSUE THREAD

**Issue by AT-Mneme** [jan-vince/smallcontactform#146]
very well-written and structured plugin! It is simple, easy setup. 

I have two questions here: 
1. I have done all setup. email template test message sent out ok, but the filled-in contact form doesn't send out to the admin email. Also, no records at contact form status list at backend. 
2. if I add the new fields like company name, or checkbox option, do I have to change setting or add in field name in smallcontactform.php; and also for the email templates?

**Comment by jan-vince** [comment]
Hi and thank you!

ad 1) Should work out of the box - if you install clean plugin, allow (in plugin Settings) option Email > Allow autoreply (email for form sender) and Email > Allow notifications (email for system admin). Screen is attached.

ad 2) No need to change anything in code.

<img width="1013" alt="Snímek obrazovky 2022-08-25 v 12 21 20" src="https://user-images.githubusercontent.com/1104383/186640766-c9b59358-3128-4620-a499-42e1280f7975.png">

**Comment by AT-Mneme** [comment]
1. installed it as clean plugin(not bundle) via October plugin. did the same as you did in screenshot. But no response. 
2. I left only even name, email and message field only. still no receiving emails from contact form input. 

Should I reinstall the plug in and re-do everything?

**Comment by jan-vince** [comment]
Look in system log for any errors. What OC version do you use?

**Comment by AT-Mneme** [comment]
removed. re-installed. having this bug. I always have this problem whenever editing/adding/removing and save page. Not only small contact form plugin. 
![Screenshot 2022-08-25 at 7 00 09 PM](https://user-images.githubusercontent.com/110415107/186648426-8a36fe21-54d7-4fb2-95e6-e573eb5e8672.png)

**Comment by jan-vince** [comment]
Try to google your error code (403 Forbidden):

Maybe this? :  https://github.com/octobercms/october/issues/5286

**Comment by AT-Mneme** [comment]
thank you so much! I think this happens to namecheap hosting too.

**Comment by AT-Mneme** [comment]
done as said for plugin. still the same-test msg ok, but plugin backend list not shown and email. This is the site: www.mne-me.com

**Comment by AT-Mneme** [comment]
> Look in system log for any errors. What OC version do you use?

475 build

**Comment by jan-vince** [comment]
Can you post screenshot of plugins settings tabs Form and Email?

**Comment by AT-Mneme** [comment]
sure. 
![Screenshot 2022-08-29 at 8 11 28 PM](https://user-images.githubusercontent.com/110415107/187198330-94e9f3f0-02a8-4503-b639-4644b444f9cc.png)
![Screenshot 2022-08-29 at 8 11 47 PM](https://user-images.githubusercontent.com/110415107/187198338-4d875114-77a2-4fb8-89c2-50b33650f746.png)

**Comment by jan-vince** [comment]
Try to uncheck `Enable AJAX` and check error messages (if any) after form is sent.

**Comment by AT-Mneme** [comment]
tried. the page just refreshed. but no form success or error message.(they haven't been appearing) still the same.

**Comment by AT-Mneme** [comment]
I always have this when having submitted the form. 
![Screenshot 2022-08-29 at 10 15 00 PM](https://user-images.githubusercontent.com/110415107/187222159-b7d10c88-d3e2-4822-af77-1c6be5d69199.png)

**Comment by jan-vince** [comment]
Looks like you have some validation error on `Your Name` field.

Small Contact form was built for Bootstrap 3 so if you use newer version, error messages can be hidden (but strange you don't see error message above your form).

Look at the field validation rules and disable them temporarily.

Maybe try to render your form on empty page without any CSS styles.

**Comment by AT-Mneme** [comment]
only field validation for name with problem. email validation no problem. removed. the problem still exists. 

I put the plugin component on partials. Does it affect that? I added a blank web page on pages with small contact form component plainly. the contact form details sent through with service part been trimmed? 

![screenshot-contact-sent](https://user-images.githubusercontent.com/110415107/187267875-ebe4a221-9906-4b5f-ad49-b55926809a2b.png)

**Comment by jan-vince** [comment]
You can put `{% component 'contactForm' %}` on partial, but plugin definition must be in layout or CMS page.

See https://docs.octobercms.com/2.x/cms/layouts.html#layout-execution-life-cycle

**Comment by AT-Mneme** [comment]
> You can put `{% component 'contactForm' %}` on partial, but plugin definition must be in layout or CMS page.

I moved the code(partials) from layout to CMS page. And put the plugin definition
JanVince\SmallContactForm\Components\SmallContactForm contactForm]
send_btn_label = "Submit"
form_success_msg = "Message Has been Successfully Sent"
form_error_msg = "Message Not Sent Out"
notification_address_to = "start@mne-me.com"
allow_redirect = 1 
in CMS page. 

It turns out the Ajax came up. but there is another error appears. 
![Screenshot 2022-08-30 at 6 31 35 PM](https://user-images.githubusercontent.com/110415107/187437317-3012514f-d3b3-4f15-8011-21f08da7ef00.png)
![Screenshot 2022-08-30 at 8 14 25 PM](https://user-images.githubusercontent.com/110415107/187468148-57e68d5b-c91b-485e-8b90-148c8a5daab9.png)

**Comment by AT-Mneme** [comment]
> Looks like you have some validation error on `Your Name` field.
> 
> Small Contact form was built for Bootstrap 3 so if you use newer version, error messages can be hidden (but strange you don't see error message above your form).
> 
> Look at the field validation rules and disable them temporarily.
> 
> Maybe try to render your form on empty page without any CSS styles.

bootstrap 4.5.

**Comment by jan-vince** [comment]
Just delete all validations on all your fields and try to send form. If this works, add validation rules one by one to see, where the problem is.

**Comment by AT-Mneme** [comment]
> Just delete all validations on all your fields and try to send form. If this works, add validation rules one by one to see, where the problem is.

that error appeared when all validations removed.

**Comment by jan-vince** [comment]
Sorry @AT-Mneme I probably can't help you :(

I use this plugin on OC1 so there can be some errors with newer version. When I have some spare time I will try it in OC2 and OC3.

Maybe try to start from scratch - use a Trash can icon in right side of plugin Settings page. Then create only one field without any validation and try to send.

**Comment by AT-Mneme** [comment]
> Sorry @AT-Mneme I probably can't help you :(
> 
> I use this plugin on OC1 so there can be some errors with newer version. When I have some spare time I will try it in OC2 and OC3.
> 
> Maybe try to start from scratch - use a Trash can icon in right side of plugin Settings page. Then create only one field without any validation and try to send.

How do we differentiate and check if it is 1.0 2.0 and 3.0? I think I have been using 1.0. I tried searching from composer.json file. It said mine is 1.0-dev.

**Comment by jan-vince** [comment]
Just look in Settings > Updates & Plugins. There is a version number in the header.

**Comment by AT-Mneme** [comment]
It is version OC1.

**Comment by jan-vince** [comment]
Should work, I have 60+ web sites using this plugin.

Can you try install Small Contact Form on clean OC installation and test it with just one basic field?

**Comment by AT-Mneme** [comment]
Yes. tried on localhost on clean oc installation and test with one field, it works perfectly. Tested on one by one. the fatal validator.php issue seems to be the Message textarea field. When you did not put any validation rule, the message textarea caused the fatal error. I put these two, and the message textarea is working well. 
* Use validation rule: `custom_not_regex`
* Use validation: `/(http|https|ftp|ftps)\:\/\/?/`
And it also seems like whenever the fill-in details failed validation, the message(at backend list) can't be received. 

I deleted the whole plugin folder. then reinstall. It works. 
I need to figure out on how to output the checkbox. 4 options of checkbox of service, it showed service "on", did not show up the checked option(s).

Thank you so much. It has been solved!
